### ISSUE TITLE
🐅 Epic: Adding support for Authentication in Ersilia

### ISSUE THREAD

**Issue by DhanshreeA** [ersilia-os/ersilia#1147]
### Summary

Some models such as Fasmifra (eos4qda) benefit from additional credentials - eg eos4qda requires sudo credentials when run from source during fetch time. We anticipate the nature and scope of these requirements to expand in the future, for example we may need API keys from external vendors for some models and we need to maintain this information in the form of metadata.

**Comment by DhanshreeA** [comment]
Related to https://github.com/ersilia-os/eos4qda/issues/2#issuecomment-2144831029

**Comment by GemmaTuron** [comment]
Hi @DhanshreeA 

Can you summarise here if there have been more discussions and plan around this?

**Comment by DhanshreeA** [comment]
There has been a little development around this and @miquelduranfrigola started work on having environment variables for storing API keys for example for certain models, proposing to add dotenv to ersilia as a dependency. However we need to revisit this.

**Comment by miquelduranfrigola** [comment]
Yes indeed - we need to make the necessary planning to push the authentication functionality forward. Not an immediate need but an important feature as we start including more and more remote API services such as ultra-large scale search or even LLMs.

As for `dotenv`, I am tempted to suggest that we implement this functionality in pure-python ourselves.

**Comment by miquelduranfrigola** [comment]
Hi @DhanshreeA

I have created an [authentication-for-models](https://github.com/ersilia-os/ersilia/tree/authentication-for-models) branch containing some initial code going in this direction. I have not made a PR since none of it is functional yet.

However, depending on the roadmap, we can take it from here in due course.

**Comment by DhanshreeA** [comment]
Sounds good, thanks @miquelduranfrigola

**Comment by GemmaTuron** [comment]
We have paused the development of this feature for the moment.
